
# Pre entrega - Python
# ALUMNA: Adriana A. Cruz - DNI 26689598




#1. **Se Crea este Archivo Colaboratory y se importan librerías necesarias.**

In [133]:
# Se importan las librerias que vamos a utilizar
import pandas as pd
import numpy as np
import os
from datetime import datetime

# **Se Monta la carpeta Drive**


In [134]:
from google.colab import drive
drive.mount('/content/drive')

#se utilizará para guardar los datos limpios
ruta_base = "/content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics"

# Mostrar todas las rutas y subrutas dentro de la carpeta base
print(f"Contenido completo de la carpeta base:\n{ruta_base}\n")

for raiz, carpetas, archivos in os.walk(ruta_base):
    print(f"📂 Carpeta: {raiz}")
    for carpeta in carpetas:
        print(f"   ├── 📁 {carpeta}")
    for archivo in archivos:
        print(f"   ├── 📄 {archivo}")
    print()

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).
Contenido completo de la carpeta base:
/content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics

📂 Carpeta: /content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics
   ├── 📁 originales
   ├── 📄 CRUZ pre_entrega.ipynb

📂 Carpeta: /content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics/originales
   ├── 📄 ventas.csv
   ├── 📄 marketing.csv
   ├── 📄 clientes.csv



Directorios de la Carpeta de Adriana Cruz

si bien se trabajo originalmente con los archivos desde el content, luego cambie a que se trabaje desde link compartido, asi siempre se puede utilizar, aunque no este logueada en el drive

carpeta de los archivos de datos originales:
> /content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics./originales

Se decide cargar con link para que el Colaboratory funcione sin problemas si lo quieren ejecutar para analizarlo.

La carpeta base se va a utilizar luego, para grabar los reportes de salida

# **CARGA DE DATOS ORIGINALES COMO DATASET**

Decido cargar con link compartido para que el ipynb de Colab funcione sin problemas si lo quieren ejecutar.

In [135]:

#pd.read_csv() - Se utiliza el link directo al archivo en mi carpeta con la opcion de bajar no de preview

# CARGA CLIENTES
df_clientes = pd.read_csv('https://drive.google.com/uc?export=download&id=1R7DHLBnnlHd0XpYskw_SAxN0BS4Dc9hu')

# CARGA VENTAS
df_ventas = pd.read_csv('https://drive.google.com/uc?export=download&id=1JCKfkpHACFNxD3dhQgGcDjiR4a_cWFwG')

# CARGA MARKETING
df_mktg = pd.read_csv('https://drive.google.com/uc?export=download&id=12jy0d-JckLd474JbdkbvOnJ_f4A572a-')

El enunciado del ejercicio pide:

2. Realizar un script básico que calcule las ventas mensuales utilizando variables y operadores.

3. Estructuras de Datos: Desarrollar un programa que almacene los datos de
ventas (producto, precio, cantidad). Decidir si conviene utilizar diccionarios o
listas.

**Estos puntos fueron eliminados durante la clase after class de los requisitos a realizar, sin embargo, se considerarán más adelante, luego de realizar el ETL**


#  **4- Introducción a Pandas: realizar un análisis exploratorio inicial de los DataFrames.**

##*DF DE CLIENTES*

In [136]:

print("Análisis exploratorio de df de Clientes:")

print("Primeras filas Clientes:")
print(df_clientes.head(5))

print("\nÚltimas filas:")
print(df_clientes.tail())

print("\nDimensiones (filas, columnas):", df_clientes.shape)

print("\nInformación del DataFrame:")
print(df_clientes.info())

print("\nEstadísticas descriptivas:")
print(df_clientes.describe())

print("\nTipos de datos:")
print(df_clientes.dtypes)

print("\n Cuenta Valores encubiertos que pueden ser nulos:")
nulos = df_clientes.apply(
    lambda x:  x.isin(['', ' ', 'NA', 'NaN', 'nan', 'NULL']).sum()
)
print(nulos)

print("\n Duplicados considerando todas las columnas:", df_clientes.duplicated().sum())

print("\n Duplicados considerando todas las columnas excepto id_cliente:", df_clientes.duplicated(subset=['nombre','edad','ciudad','ingresos']).sum())

Análisis exploratorio de df de Clientes:
Primeras filas Clientes:
   id_cliente               nombre  edad         ciudad  ingresos
0           1      Aloysia Screase    44  Mar del Plata  42294.68
1           2  Kristina Scaplehorn    25        Posadas  24735.04
2           3       Filip Castagne    50    Resistencia  35744.85
3           4          Liuka Luard    39   Bahía Blanca  27647.96
4           5        Dore Cockshtt    28        Rosario  28245.65

Últimas filas:
     id_cliente           nombre  edad       ciudad  ingresos
562         563    Dione Forsyde    29      Posadas  26757.73
563         564      Fleming Gow    39     Santa Fe  43674.96
564         565  Jewelle Mabbett    33      Córdoba  30522.64
565         566      Lauri Munns    23  Resistencia  31259.14
566         567      Micah Matis    31   Corrientes  42927.86

Dimensiones (filas, columnas): (567, 5)

Información del DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 567 entries, 0 to 566
Data colu

**Ideas del analisis exploratorio sobre CLIENTES:**
1. Estructura general
El DataFrame contiene 567 registros y 5 columnas: id_cliente, nombre, edad, ciudad e ingresos.
El tipo de datos de cada campo es correcto: identificador y edad como enteros, ingresos como número decimal, y nombre y ciudad como texto.

2. Completitud de los datos
No existen valores nulos ni vacíos encubiertos (0 valores en todas las columnas).

3. Duplicados
No se detectan registros duplicados ni considerando todas las columnas ni excluyendo el identificador (id_cliente).
La integridad referencial de los datos se mantiene correctamente.

4. Estadísticas descriptivas
Edad: promedio 37,9 años, con un rango entre 20 y 81: población adulta.

Ingresos: promedio de 34.668,7 con una desviación de 12.974,5. El rango va de **170,29** a 88.053,01,  **hay riesgo de posibles valores atípicos bajos.**

Id_cliente: secuencia numérica completa de 1 a 567, sin interrupciones.

# DF DE MKTG
Analisis exploratorio

In [137]:
print("Análisis exploratorio de df de Marketing:")

print("Primeras filas Marketing:")
print(df_mktg.head())

print("\nÚltimas filas:")
print(df_mktg.tail())

print("\nDimensiones (filas, columnas):", df_mktg.shape)

print("\nInformación del DataFrame:")
print(df_mktg.info())

print("\nEstadísticas descriptivas:")
print(df_mktg.describe())

print("\nTipos de datos:")
print(df_mktg.dtypes)

print("\n Cuenta Valores encubiertos que pueden ser nulos:")
nulos = df_mktg.apply(
    lambda x:  x.isin(['', ' ', 'NA', 'NaN', 'nan', 'NULL']).sum()
)
print(nulos)

print("\n Duplicados considerando todas las columnas:", df_mktg.duplicated().sum())

print("\n Duplicados considerando todas las columnas excepto id_campanha:", df_mktg.duplicated(subset=['producto','canal','costo','fecha_inicio',"fecha_fin"]).sum())

Análisis exploratorio de df de Marketing:
Primeras filas Marketing:
   id_campanha         producto  canal  costo fecha_inicio   fecha_fin
0           74  Adorno de pared     TV   4.81   20/03/2024  03/05/2024
1           12           Tablet   RRSS   3.40   26/03/2024  13/05/2024
2           32  Lámpara de mesa  Email   5.54   28/03/2024  20/04/2024
3           21       Smartphone   RRSS   6.37   29/03/2024  16/05/2024
4           58         Alfombra  Email   4.25   31/03/2024  05/05/2024

Últimas filas:
    id_campanha            producto  canal  costo fecha_inicio   fecha_fin
85           70          Aspiradora     TV   3.06   13/12/2024  29/12/2024
86           89           Televisor     TV   4.98   13/12/2024    8/2/2025
87           68   Rincón de plantas     TV   5.81   17/12/2024   14/2/2025
88           33            Secadora  Email   3.80   20/12/2024    7/1/2025
89           11  Freidora eléctrica   RRSS   5.27   29/12/2024   21/1/2025

Dimensiones (filas, columnas): (90, 6)


## **IDEAS de la Exploración en la DF de MKTG**

1. Estructura general
El DataFrame de marketing contiene 90 registros y 6 columnas: id_campanha, producto, canal, costo, fecha_inicio y fecha_fin.
Los tipos de datos son consistentes: identificador numérico, variables categóricas (producto, canal), variable numérica (costo) y fechas representadas como texto.

2. Completitud de los datos
No se detectan valores nulos ni encubiertos (0 en todas las columnas).

3. Duplicados
No existen registros duplicados, ni considerando todas las columnas ni excluyendo el identificador id_campanha.

4. Estadísticas descriptivas
id_campanha:  codificación secuencial.
costo: media de 4,93 con una desviación estándar de 0,95.
Los valores oscilan entre 2,95 y 7,39, mostrando un rango moderado y homogeneo

5.  Variables de fecha
Las columnas fecha_inicio y fecha_fin están almacenadas como texto (object).
Los formatos no son homogéneos: se observan valores con días y meses de una sola cifra (ej. 8/2/2025), lo cual puede generar errores de interpretación.
Será necesario estandarizar formato y convertirlas a datetime para análisis de duración o calendario de campañas.

6. **El formato de fechas es el único aspecto a normalizar antes de proceder a análisis o modelado.**

#DF de VENTAS
Análisis Exploratorio

In [138]:
print("Análisis exploratorio de df de Ventas:")

print("Primeras filas Ventas:")
print(df_ventas.head())

print("\nÚltimas filas:")
print(df_ventas.tail())

print("\nDimensiones (filas, columnas):", df_ventas.shape)

print("\nInformación del DataFrame:")
print(df_ventas.info())

print("\nEstadísticas descriptivas:")
print(df_ventas.describe())

print("\nTipos de datos:")
print(df_ventas.dtypes)

print("\n Cuenta Valores encubiertos que pueden ser nulos:")
nulos = df_ventas.apply(
    lambda x:  x.isin(['', ' ', 'NA', 'NaN', 'nan', 'NULL']).sum()
)
print(nulos)

print("\n Duplicados considerando todas las columnas:", df_ventas.duplicated().sum())

print("\n Duplicados considerando todas las columnas excepto id_venta:", df_ventas.duplicated(subset=['producto','precio','cantidad','fecha_venta']).sum())


Análisis exploratorio de df de Ventas:
Primeras filas Ventas:
   id_venta           producto   precio  cantidad fecha_venta  \
0       792  Cuadro decorativo   $69.94       5.0  02/01/2024   
1       811    Lámpara de mesa  $105.10       5.0  02/01/2024   
2      1156           Secadora   $97.96       3.0  02/01/2024   
3      1372           Heladera  $114.35       8.0  02/01/2024   
4      1546           Secadora  $106.21       4.0  02/01/2024   

           categoria  
0         Decoración  
1         Decoración  
2  Electrodomésticos  
3  Electrodomésticos  
4  Electrodomésticos  

Últimas filas:
      id_venta                producto   precio  cantidad fecha_venta  \
3030      1837         Horno eléctrico  $104.12       9.0  30/12/2024   
3031      2276                  Laptop   $85.27       9.0  30/12/2024   
3032      2696                  Laptop  $107.81       4.0  30/12/2024   
3033      2913              Smartphone   $99.85       7.0  30/12/2024   
3034      2930  Consola de v

# **Ideas sobre el estado de la base de VENTAS**
1. Estructura general
El DataFrame de ventas contiene 3035 registros y 6 columnas: id_venta, producto, precio, cantidad, fecha_venta y categoria.
El tipo de datos es mayormente correcto, aunque se observa que **precio está almacenado como texto (con símbolo $)** en lugar de valor numérico, y que fecha_venta aún no fue convertida a formato de fecha.

2. Completitud de los datos
id_venta, producto, fecha_venta y categoria no presentan nulos.
Se detectan 2 valores nulos en precio y cantidad, ya que el conteo es de 3033 frente a 3035 filas.
El conteo de “valores encubiertos” devuelve 0, pero el .info() confirma estas ausencias reales.

3. Duplicados
**Se detectan 35 duplicados exactos**

4. Estadísticas descriptivas

Cantidad: media de 6,5 unidades por venta, con una desviación de 3,45. El rango (1–12) no parecen valores extremos.

id_venta: numérico y consecutivo de 1 a 3000, aunque existen 3035 filas, lo que sugiere que los identificadores pueden haberse duplicado o generado en distintos orígenes.
Precio: no incluido en las estadísticas por estar en formato texto. Para análisis económico, es necesario transformarlo a numérico.

5. Variables de fecha
fecha_venta está en formato object, con valores tipo "dd/mm/yyyy".
Se convertira a datetime


# 5. Calidad de Datos: Identificar valores nulos y duplicados en los conjuntos de datos. Documentar el estado inicial de los datos.

Segun vimos en el analisis exploratorio de las tres bases analizadas, Clientes y Marketing se encuentran limpias, sin valores nulos ni nulos "encubiertos" en otras formas de registro y sin registros duplicados, con estructuras coherentes y tipos de datos consistentes.

 En cambio, **la base de Ventas presenta algunos problemas de calidad: existen valores nulos reales en las columnas precio y cantidad, y además se detectaron 35 registros duplicados** (tanto considerando todas las columnas como excluyendo el identificador id_venta). En consecuencia, solo el dataset de Ventas requiere tareas de limpieza y normalización, mientras que las bases de Clientes y Marketing pueden considerarse listas para su integración y análisis.

In [139]:
#Se muestran los casos de DUPLICADOS
# DUPLICADOS EN CLIENTES - NO HAY
# ============================================================
print("\n DUPLICADOS EN CLIENTES")
print(df_clientes[df_clientes.duplicated(keep=False)])


# DUPLICADOS EN MARKETING - NO HAY
# ============================================================
print("\n DUPLICADOS EN MARKETING")
print(df_mktg[df_mktg.duplicated(keep=False)])


# DUPLICADOS EN VENTAS - Si HAY segun se vio en la exploracion
# ============================================================

#solo mostrar combinaciones repetidas
print("\nDuplicados en VENTAS (solo mostrar combinaciones repetidas------ CANTIDAD de VECES REPETIDA):")
print(
    df_ventas.groupby(['producto','precio','cantidad','fecha_venta','categoria'])
    .size()[lambda x: x > 1]
)


 DUPLICADOS EN CLIENTES
Empty DataFrame
Columns: [id_cliente, nombre, edad, ciudad, ingresos]
Index: []

 DUPLICADOS EN MARKETING
Empty DataFrame
Columns: [id_campanha, producto, canal, costo, fecha_inicio, fecha_fin]
Index: []

Duplicados en VENTAS (solo mostrar combinaciones repetidas------ CANTIDAD de VECES REPETIDA):
producto                precio   cantidad  fecha_venta  categoria        
Adorno de pared         $67.75   9.0       06/04/2024   Decoración           2
Alfombra                $41.72   8.0       06/04/2024   Decoración           2
                        $56.78   9.0       07/04/2024   Decoración           2
Aspiradora              $115.24  3.0       06/04/2024   Electrodomésticos    2
                        $87.32   7.0       06/04/2024   Electrodomésticos    2
Auriculares             $122.56  6.0       07/04/2024   Electrónica          2
                        $32.81   11.0      05/04/2024   Electrónica          2
                        $87.29   2.0       08/04/

In [140]:
## Se cuentan y  muestran los casos de Valores Nulos
print("\n=== VALORES NULOS EN CLIENTES === NO HAY")
print(df_clientes.isnull().sum())
print("\nFilas con valores nulos en CLIENTES:")
print(df_clientes[df_clientes.isnull().any(axis=1)])

print("\n=== VALORES NULOS EN MARKETING === NO HAY")
print(df_mktg.isnull().sum())
print("\nFilas con valores nulos en MARKETING:")
print(df_mktg[df_mktg.isnull().any(axis=1)])

print("\n=== VALORES NULOS EN VENTAS ===")
print(df_ventas.isnull().sum())
print("\nFilas con valores nulos en VENTAS:")
print(df_ventas[df_ventas.isnull().any(axis=1)])


=== VALORES NULOS EN CLIENTES === NO HAY
id_cliente    0
nombre        0
edad          0
ciudad        0
ingresos      0
dtype: int64

Filas con valores nulos en CLIENTES:
Empty DataFrame
Columns: [id_cliente, nombre, edad, ciudad, ingresos]
Index: []

=== VALORES NULOS EN MARKETING === NO HAY
id_campanha     0
producto        0
canal           0
costo           0
fecha_inicio    0
fecha_fin       0
dtype: int64

Filas con valores nulos en MARKETING:
Empty DataFrame
Columns: [id_campanha, producto, canal, costo, fecha_inicio, fecha_fin]
Index: []

=== VALORES NULOS EN VENTAS ===
id_venta       0
producto       0
precio         2
cantidad       2
fecha_venta    0
categoria      0
dtype: int64

Filas con valores nulos en VENTAS:
     id_venta               producto precio  cantidad fecha_venta    categoria
136       627  Elementos de cerámica    NaN       NaN  17/01/2024   Decoración
139      2171    Parlantes Bluetooth    NaN       NaN  17/01/2024  Electrónica


# ETAPA 2. Preprocesamiento y Limpieza de datos


---


# 1. Limpieza de datos.

---


1.  Eliminación de **DUPLICADOS**
2. Convertimos los valores que deberian ser numéricos, para poder tener datos estadisticos (precio)
3.  Tratamiento de los Valores **NULOS**. Criterios de decisión, estimacione, Debemos tomar decision si los estimamos acercándolos a alguna medida central apropiada o eliminamos los registros. - Decido:** Eliminar**
4. Pasamos a valor fecha, los valores Object que representan fechas.
Pasamos a valor fecha, los valores Object.  
5.   Normalización de textos (de valores Object que son string), formato "Titulo", primeras mayúsculas





In [141]:
#Realizaremos la limpieza sobre copias de las bases
ventas_etl = df_ventas.copy()
clientes_etl = df_clientes.copy()
mktg_etl= df_mktg.copy()


## Eliminación de los Duplicados
Los valores duplicados afectan la calidad del analisis posterior.

la unica base con datos duplicados era VENTAS.  Siendo que hay casos donde estan duplicados hasta con el ID, no hay duda que hay que eliminarlos, ya que al tener el msimo ID sabemos que no son operaciones casualmente con los mismos valores.



In [142]:
## Ventas tenia duplicados
print("\n VENTAS - LA BASE ANTES: Duplicados considerando todas las columnas:", ventas_etl.duplicated().sum())
ventas_etl=ventas_etl.drop_duplicates()
print("VENTAS - LA BASE DESPUES de hacer DROP: Duplicados considerando todas las columnas:", ventas_etl.duplicated().sum())

##No tenian duplicados
print("\n CLIENTES- LA BASE ANTES: Duplicados considerando todas las columnas:", clientes_etl.duplicated().sum())
clientes_etl=clientes_etl.drop_duplicates()
print("CLIENTES -LA BASE DESPUES de hacer DROP: Duplicados considerando todas las columnas:", clientes_etl.duplicated().sum())

print("\n LA BASE ANTES: Duplicados considerando todas las columnas:", mktg_etl.duplicated().sum())
mktg_etl=mktg_etl.drop_duplicates()
print("LA BASE DESPUES de hacer DROP: Duplicados considerando todas las columnas:", mktg_etl.duplicated().sum())




 VENTAS - LA BASE ANTES: Duplicados considerando todas las columnas: 35
VENTAS - LA BASE DESPUES de hacer DROP: Duplicados considerando todas las columnas: 0

 CLIENTES- LA BASE ANTES: Duplicados considerando todas las columnas: 0
CLIENTES -LA BASE DESPUES de hacer DROP: Duplicados considerando todas las columnas: 0

 LA BASE ANTES: Duplicados considerando todas las columnas: 0
LA BASE DESPUES de hacer DROP: Duplicados considerando todas las columnas: 0


## Conversión de columna Precio a Numérica

In [143]:
# Convertimos la columna 'precio' a numérica (quitando el símbolo $)
ventas_etl['precio'] = ventas_etl['precio'].replace('[\\$,]', '', regex=True).astype(float)

## ¿Que hacemos con los valores Nulos? Hay forma de estimarlos?



In [144]:

print("\n=== VALORES NULOS EN VENTAS ===")
print(ventas_etl.isnull().sum())
print("\nFilas con valores nulos en VENTAS:")
print(ventas_etl[ventas_etl.isnull().any(axis=1)])


=== VALORES NULOS EN VENTAS ===
id_venta       0
producto       0
precio         2
cantidad       2
fecha_venta    0
categoria      0
dtype: int64

Filas con valores nulos en VENTAS:
     id_venta               producto  precio  cantidad fecha_venta  \
136       627  Elementos de cerámica     NaN       NaN  17/01/2024   
139      2171    Parlantes Bluetooth     NaN       NaN  17/01/2024   

       categoria  
136   Decoración  
139  Electrónica  


In [145]:
# Mostramos estadísticas de PRECIO
print("\n=== ESTADÍSTICAS DE PRECIO (sin valores nulos) ===")
print("Mínimo:", ventas_etl['precio'].dropna().min())
print("Máximo:", ventas_etl['precio'].dropna().max())
print("Media:", ventas_etl['precio'].dropna().mean())
print("Mediana:", ventas_etl['precio'].dropna().median())

# Mostramos estadísticas de CANTIDAD
print("\n=== ESTADÍSTICAS DE CANTIDAD (sin valores nulos) ===")
print("Mínimo:", ventas_etl['cantidad'].dropna().min())
print("Máximo:", ventas_etl['cantidad'].dropna().max())
print("Media:", ventas_etl['cantidad'].dropna().mean())
print("Mediana:", ventas_etl['cantidad'].dropna().median())


# Identificamos los productos con nulos en precio o cantidad
productos_con_nulos = ventas_etl.loc[
    ventas_etl['precio'].isnull() | ventas_etl['cantidad'].isnull(),
    'producto'
].unique()

# Mostramos sus estadísticas (solo sobre los valores válidos)
print("\n=== ESTADÍSTICAS DE PRODUCTOS CON NULOS EN PRECIO O CANTIDAD ===")
print(
    ventas_etl[ventas_etl['producto'].isin(productos_con_nulos)]
    .groupby('producto')[['precio','cantidad']]
    .agg(['min','max','mean','median'])
)



=== ESTADÍSTICAS DE PRECIO (sin valores nulos) ===
Mínimo: 26.0
Máximo: 124.97
Media: 75.29131754503001
Mediana: 75.205

=== ESTADÍSTICAS DE CANTIDAD (sin valores nulos) ===
Mínimo: 1.0
Máximo: 12.0
Media: 6.5026684456304205
Mediana: 7.0

=== ESTADÍSTICAS DE PRODUCTOS CON NULOS EN PRECIO O CANTIDAD ===
                      precio                           cantidad        \
                         min     max       mean median      min   max   
producto                                                                
Elementos de cerámica  27.43  124.38  71.611515  66.54      1.0  12.0   
Parlantes Bluetooth    26.24  124.34  73.583571  76.53      1.0  12.0   

                                        
                           mean median  
producto                                
Elementos de cerámica  6.424242    6.0  
Parlantes Bluetooth    6.734694    7.0  


## Decisión sobre valores Nulos

Solo 2 registros nulos en precio y cantidad (id_venta 627 y 2171).

Ambos pertenecen a productos que sí tienen suficientes observaciones válidas (Elementos de cerámica y Parlantes Bluetooth), con medias y medianas razonablemente estables.

El resto de la base está limpia, con más de 3000 filas.

Consideramos
Los nulos no son representativos, pero sí podrían distorsionar cálculos agregados (por ejemplo, ventas totales, promedios por producto, etc.).

Dado el volumen del dataset, eliminar 2 filas de 3035 no afecta la representatividad estadística.Dado que son solo 2 casos de 3035 (0,066%).

##***Decido: Eliminar esos dos registros.***
No justifican una imputación estadística ni aportan información adicional.
La imputación podría generar una falsa sensación de precisión en los valores de venta.

In [146]:
# Eliminar registros con nulos en precio o cantidad
print("Filas antes de eliminar:", ventas_etl.shape[0])
ventas_etl.dropna(subset=['precio', 'cantidad'], inplace=True)
print("Filas después de eliminar:", ventas_etl.shape[0])

Filas antes de eliminar: 3000
Filas después de eliminar: 2998


## Convertimos el tipo de dato de los valores de fechas que figuran en el dataframe como Object

In [147]:
# Normalizar fechas
# -------------------------------------------------
# Si alguna columna contiene fechas  se intenta convertir a formato datetime de pandas.
# to_datetime intenta interpretar el formato y transforma valores inválidos en NaT (Not a Time).

for nombre_df, df in {'ventas_etl': ventas_etl, 'clientes_etl': clientes_etl, 'mktg_etl': mktg_etl}.items():
    try:
        for col in df.columns:
            if "fecha" in col.lower():
                try:
                    df[col] = pd.to_datetime(df[col], errors="coerce", dayfirst=True)
                    print(f" {nombre_df}: columna '{col}' convertida correctamente.")
                except Exception as e:
                    print(f"  Error convirtiendo columna '{col}' en {nombre_df}: {e}")
    except Exception as e:
        print(f"  Error procesando {nombre_df}: {e}")


## verificamos años con datos de venta

# Extraemos el año
ventas_etl['anio'] = ventas_etl['fecha_venta'].dt.year

# Listamos los años detectados
anios_presentes = ventas_etl['anio'].dropna().unique()
anios_presentes.sort()

print("=== AÑOS DETECTADOS EN LA BASE DE VENTAS ===")
print(anios_presentes)


 ventas_etl: columna 'fecha_venta' convertida correctamente.
 mktg_etl: columna 'fecha_inicio' convertida correctamente.
 mktg_etl: columna 'fecha_fin' convertida correctamente.
=== AÑOS DETECTADOS EN LA BASE DE VENTAS ===
[2024]


In [148]:
# === NORMALIZACIÓN DE CAMPOS DE TEXTO ===

dataframes = [clientes_etl, ventas_etl, mktg_etl]

for df in dataframes:
    for col in df.select_dtypes(include=['object']).columns:
        # Normaliza texto: primera letra mayúscula, resto minúsculas
        df[col] = df[col].astype(str).str.strip().str.title()

print("Normalización completada: todas las columnas de texto tienen formato 'Primera Letra En Mayúscula'.")


Normalización completada: todas las columnas de texto tienen formato 'Primera Letra En Mayúscula'.


##Estructuras de Datos. Se desarrolla el script para que almacene los datos de las bases limpias, incluyendo  los datos de ventas.  No se requiere usar ni listas ni diccionarios.

Utiliza Google Drive, montado previamente y la carpeta base
(corresponde a la Etapa 1, item 3 del enunciado)


In [149]:

# . Crear una subcarpeta específica para este ETL
#
nombre_subcarpeta = "Datos Limpios"
ruta_destino = os.path.join(ruta_base, nombre_subcarpeta)
os.makedirs(ruta_destino, exist_ok=True)

print(f"Carpeta destino creada/verificada: {ruta_destino}")

# Graba los archivos
ventas_etl.to_csv(os.path.join(ruta_destino, "ventas_etl.csv"), index=False, encoding="utf-8")
clientes_etl.to_csv(os.path.join(ruta_destino, "clientes_etl.csv"), index=False, encoding="utf-8")
mktg_etl.to_csv(os.path.join(ruta_destino, "mktg_etl.csv"), index=False, encoding="utf-8")

print("\nArchivos ETL guardados correctamente en Google Drive:")
print(f" - {os.path.join(ruta_destino, 'ventas_etl.csv')}")
print(f" - {os.path.join(ruta_destino, 'clientes_etl.csv')}")
print(f" - {os.path.join(ruta_destino, 'mktg_etl.csv')}")

Carpeta destino creada/verificada: /content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics/Datos Limpios

Archivos ETL guardados correctamente en Google Drive:
 - /content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics/Datos Limpios/ventas_etl.csv
 - /content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics/Datos Limpios/clientes_etl.csv
 - /content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics/Datos Limpios/mktg_etl.csv


In [150]:
# Correlación entre precio y cantidad
corr = ventas_etl['precio'].corr(ventas_etl['cantidad'])
print(f"Correlación entre PRECIO y CANTIDAD: {corr:.4f}")

analisis_por_producto = (
    ventas_etl.groupby('producto')[['precio', 'cantidad']]
    .corr().unstack().iloc[:, 1]  # correlación entre precio y cantidad
)
print(analisis_por_producto)


Correlación entre PRECIO y CANTIDAD: -0.0024
producto
Adorno De Pared          -0.009250
Alfombra                 -0.072439
Aspiradora               -0.036100
Auriculares               0.077525
Batidora                 -0.122274
Cafetera                 -0.066740
Candelabro               -0.328097
Consola De Videojuegos   -0.155294
Cortinas                 -0.006209
Cuadro Decorativo         0.015647
Cámara Digital           -0.077943
Elementos De Cerámica    -0.012571
Espejo Decorativo        -0.121881
Freidora Eléctrica        0.090183
Heladera                  0.201254
Horno Eléctrico           0.224559
Jarrón Decorativo         0.112332
Laptop                   -0.028753
Lavadora                 -0.135006
Lámpara De Mesa           0.081650
Microondas                0.027795
Parlantes Bluetooth      -0.081175
Plancha De Vapor         -0.091487
Proyector                 0.154171
Rincón De Plantas        -0.117497
Secadora                  0.110315
Smartphone                0.000218
S

# Reportes
en esta sección se crean los reportes con análisis solicitados

## Crear subcarpeta 'reportes' si no existe

## Tareas a realizar en esta seccion

*   Realizar un script básico que calcule las ventas mensuales utilizando
variables y operadores (Punto 2 de la primera parte del enunciado)
*   Transformación de Datos: Aplicar filtros y transformaciones para crear una
tabla de ventas que muestre solo los productos con alto rendimiento. (punto 2 de la segunda parte del enunciado)
* Agregación: Resumir las ventas por categoría de producto y analizar los
ingresos generados. (punto 3 de la segunda parte)
* Integración de Datos: Combinar los sets de datos de ventas y marketing
para obtener una visión más amplia de las tendencias. (punto 4 de la segunda parte).
* Bonus Track.


In [151]:
#Crea ruta reportes si no existe.
ruta_reportes = os.path.join(ruta_base, "reportes")
os.makedirs(ruta_reportes, exist_ok=True)

## Ventas Mensuales (corresponde a la Etapa 1, item 2 del enunciado)
Ahora que tenemos el dato de fecha de ventas en el formato adecuado podemos Agregar la información para mostrarla

In [152]:
# ============================================================
# CÁLCULO DE VENTAS TOTALES MENSUALES

# Calcular el importe total por registro
ventas_etl['importe_total'] = ventas_etl['precio'] * ventas_etl['cantidad']

#  Crear columna Año-Mes
ventas_etl['anio_mes'] = ventas_etl['fecha_venta'].dt.to_period('M')

# Agrupar por mes y sumar importes
ventas_mensuales = (
    ventas_etl.groupby('anio_mes')['importe_total']
    .sum()
    .reset_index()
    .sort_values('anio_mes')
)

#  Mostrar resultados
print("=== VENTAS TOTALES MENSUALES ===")
print(ventas_mensuales)

## Guardar archivo en la subcarpeta
ventas_mensuales.to_csv(f"{ruta_reportes}/ventas_totales_mensuales.csv", index=False, encoding="utf-8")



=== VENTAS TOTALES MENSUALES ===
   anio_mes  importe_total
0   2024-01      129604.99
1   2024-02      118672.44
2   2024-03      136779.15
3   2024-04      128430.69
4   2024-05      143727.25
5   2024-06      108480.17
6   2024-07      116229.97
7   2024-08      119680.15
8   2024-09      115787.85
9   2024-10      112117.13
10  2024-11      119951.79
11  2024-12      117631.94


# Transformación de Datos (Punto 2, etapa 2)

Aplicar filtros y transformaciones para crear una tabla de ventas que muestre solo los productos con alto rendimiento.


In [153]:
# Agrupar por producto y calcular métricas relevantes
ventas_top = (
    ventas_etl.groupby('producto')
    .agg(
        importe_total=('importe_total', 'sum'),      # total vendido (precio x cantidad)
        unidades=('cantidad', 'sum'),                # total de unidades vendidas
        precio_promedio=('precio', 'mean'),          # precio promedio unitario
        registros=('producto', 'count')              # cantidad de registros
    )
    .reset_index()
    .sort_values('importe_total', ascending=False)   # orden descendente por importe
)
# Agrupar por producto y calcular métricas relevantes
ventas_top = (
    ventas_etl.groupby('producto')
    .agg(
        importe_total=('importe_total', 'sum'),      # total vendido (precio x cantidad)
        unidades=('cantidad', 'sum'),                # total de unidades vendidas
        precio_promedio=('precio', 'mean'),          # precio promedio unitario
        registros=('producto', 'count')              # cantidad de registros
    )
    .reset_index()
    .sort_values('importe_total', ascending=False)   # orden descendente por importe
)

# Mostrar productos ordenados de mayor a menor importe total
print("\n----- PRODUCTOS ORDENADOS DE MAYOR A MENOR IMPORTE DE VENTAS ------")
print(ventas_top[['producto','importe_total']])

# Calcular el valor del percentil 80 del importe total
umbral_p80 = ventas_top['importe_total'].quantile(0.8)

# Filtrar productos que superan el percentil 80
ventas_top80 = ventas_top[ventas_top['importe_total'] > umbral_p80]

# Mostrar productos del top 20 por ciento
print("\n----- TOP PERCENTIL 80 - POR IMPORTE TOTAL DE VENTAS (Precio x Cantidad) ------")
print(ventas_top80)

# ============================================================
# GUARDAR RESULTADO EN LA CARPETA DE REPORTES
# ============================================================

ventas_top80.to_csv(
    f"{ruta_reportes}/ventas_top80_completo.csv",
    index=False,
    encoding="utf-8"
)

print(f"\nArchivo generado correctamente en: {ruta_reportes}/ventas_top80_completo.csv")

# Calcular el valor del percentil 80 del importe total
umbral_p80 = ventas_top['importe_total'].quantile(0.8)

# Filtrar productos que superan el percentil 80
ventas_top80 = ventas_top[ventas_top['importe_total'] > umbral_p80]

# Mostrar productos del top 20 por ciento
print("\n----- TOP PERCENTIL 80 - POR IMPORTE TOTAL DE VENTAS (Precio x Cantidad) ------")
print(ventas_top80)

# ============================================================
# GUARDAR RESULTADO EN LA CARPETA DE REPORTES

ventas_top80.to_csv(
    f"{ruta_reportes}/ventas_top80_completo.csv",
    index=False,
    encoding="utf-8"
)

print(f"\nArchivo generado correctamente en: {ruta_reportes}/ventas_top80_completo.csv")


----- PRODUCTOS ORDENADOS DE MAYOR A MENOR IMPORTE DE VENTAS ------
                  producto  importe_total
19         Lámpara De Mesa       82276.38
3              Auriculares       74175.58
20              Microondas       72562.89
5                 Cafetera       59607.31
9        Cuadro Decorativo       54297.60
26              Smartphone       54132.44
25                Secadora       52115.45
16       Jarrón Decorativo       51130.88
4                 Batidora       50979.20
24       Rincón De Plantas       50456.45
13      Freidora Eléctrica       50155.15
2               Aspiradora       50085.86
15         Horno Eléctrico       49913.90
18                Lavadora       48946.44
23               Proyector       48187.00
0          Adorno De Pared       48093.49
27              Smartwatch       47836.86
21     Parlantes Bluetooth       47779.46
29               Televisor       47038.04
12       Espejo Decorativo       46783.31
17                  Laptop       46179.48
7   Con

**ANALISIS: **Los productos que representan el 80% de las ventas totales son: la Lámpara de mesa, los auriculares, el microondas, la cafetera, el cuadro decorativo y el smarphone, con precios unitarios entre 72,72 y 81,39 pesos.


---

# **Agregación**: Resumir las ventas por categoría de producto y analizar los ingresos generados.(ETAPA 2 ITEM 3 del enunciado)

In [154]:
# Agrupar por categoría y calcular métricas relevantes
ventas_categoria = (
    ventas_etl.groupby('categoria')
    .agg(
        importe_total=('importe_total', 'sum'),   # ingresos totales por categoría
        unidades=('cantidad', 'sum'),             # unidades totales vendidas
        precio_promedio=('precio', 'mean'),       # precio promedio unitario
        registros=('categoria', 'count')          # cantidad de transacciones
    )
    .reset_index()
    .sort_values('importe_total', ascending=False)  # orden descendente por ingresos
)

# Mostrar resultados
print("\n----- RESUMEN DE VENTAS POR CATEGORIA DE PRODUCTO ------")
print(ventas_categoria)


# Calcular participación porcentual de cada categoría
ventas_categoria['porcentaje_ingresos'] = (
    ventas_categoria['importe_total'] / ventas_categoria['importe_total'].sum() * 100
)

# Mostrar resumen con participación
print("\n----- PARTICIPACION DE CADA CATEGORIA EN LOS INGRESOS TOTALES ------")
print(ventas_categoria[['categoria', 'importe_total', 'porcentaje_ingresos']])

# GUARDAR REPORTE EN LA CARPETA DE REPORTES

ventas_categoria.to_csv(
    f"{ruta_reportes}/ventas_por_categoria.csv",
    index=False,
    encoding="utf-8"
)

print(f"\nArchivo generado correctamente en: {ruta_reportes}/ventas_por_categoria.csv")


----- RESUMEN DE VENTAS POR CATEGORIA DE PRODUCTO ------
           categoria  importe_total  unidades  precio_promedio  registros
1  Electrodomésticos      505299.63    6592.0         76.52096       1000
2        Electrónica      482577.80    6413.0         75.25492        998
0         Decoración      479216.09    6490.0         74.09800       1000

----- PARTICIPACION DE CADA CATEGORIA EN LOS INGRESOS TOTALES ------
           categoria  importe_total  porcentaje_ingresos
1  Electrodomésticos      505299.63            34.442224
2        Electrónica      482577.80            32.893459
0         Decoración      479216.09            32.664318

Archivo generado correctamente en: /content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics/reportes/ventas_por_categoria.csv


**Análisis:** Los resultados muestran una distribución equilibrada de los ingresos entre las tres categorías principales, con una leve predominancia de Electrodomésticos, que concentra el 34,4 % del total facturado.
Le siguen Electrónica (32,9 %) y Decoración (32,7 %), con diferencias menores al 2 %.

En conjunto, las tres categorías generan volúmenes de venta y precios promedio similares (entre 74 y 77 unidades monetarias), lo que sugiere una estructura de catálogo homogénea, sin dependencia significativa de un único segmento.



---


#** Integración de datos** (Etapa 2 Item 4 del enunciado):*
Combinar los sets de datos de ventas y marketing para obtener una visión más amplia de las tendencias.

Listado completo, por registro

In [155]:
import pandas as pd
import numpy as np

# === 1. Determinar el periodo total de análisis ===
inicio_periodo = ventas_etl['fecha_venta'].min()
fin_periodo = ventas_etl['fecha_venta'].max()
dias_periodo_total = (fin_periodo - inicio_periodo).days + 1

print("Periodo analizado:", inicio_periodo.date(), "a", fin_periodo.date(), f"({dias_periodo_total} días)")

# === 2. Convertir fechas si aún no están en formato datetime ===
mktg_etl['fecha_inicio'] = pd.to_datetime(mktg_etl['fecha_inicio'], dayfirst=True)
mktg_etl['fecha_fin'] = pd.to_datetime(mktg_etl['fecha_fin'], dayfirst=True)
ventas_etl['fecha_venta'] = pd.to_datetime(ventas_etl['fecha_venta'], dayfirst=True)

# === 3. Marcar ventas dentro de campaña SIN duplicar registros ===
ventas_etl['en_campania'] = False

for _, camp in mktg_etl.iterrows():
    mask = (
        (ventas_etl['producto'] == camp['producto']) &
        (ventas_etl['fecha_venta'] >= camp['fecha_inicio']) &
        (ventas_etl['fecha_venta'] <= camp['fecha_fin'])
    )
    ventas_etl.loc[mask, 'en_campania'] = True

# === 4. Separar ventas con y sin campaña ===
ventas_con_campania = ventas_etl[ventas_etl['en_campania'] == True].copy()
ventas_sin_campania = ventas_etl[ventas_etl['en_campania'] == False].copy()

print("Ventas con campaña:", len(ventas_con_campania))
print("Ventas sin campaña:", len(ventas_sin_campania))

# === 5. Calcular días de campaña por producto y canal ===
dias_campania = (
    mktg_etl
    .groupby(['producto', 'canal'])[['fecha_inicio', 'fecha_fin']]
    .apply(lambda x: (x['fecha_fin'].max() - x['fecha_inicio'].min()).days + 1)
    .reset_index(name='dias_campania')
)

# === 6. Agrupar ventas con campaña ===
con_campania = (
    ventas_con_campania
    .groupby(['producto'])
    .agg(cantidad=('cantidad', 'sum'), importe_total=('importe_total', 'sum'))
    .reset_index()
)

# Añadir canal y días de campaña desde marketing
con_campania = pd.merge(con_campania, dias_campania, on='producto', how='left')
con_campania['indice_venta_dia'] = con_campania['importe_total'] / con_campania['dias_campania']

# === 7. Calcular días fuera de campaña ===
dias_totales_prod = dias_campania.groupby('producto')['dias_campania'].sum().reset_index()
dias_totales_prod['dias_fuera_campania'] = dias_periodo_total - dias_totales_prod['dias_campania']
dias_totales_prod['dias_fuera_campania'] = dias_totales_prod['dias_fuera_campania'].clip(lower=1)

# === 8. Agrupar ventas sin campaña ===
sin_campania = (
    ventas_sin_campania
    .groupby('producto')
    .agg(cantidad=('cantidad', 'sum'), importe_total=('importe_total', 'sum'))
    .reset_index()
)
sin_campania = pd.merge(sin_campania, dias_totales_prod[['producto', 'dias_fuera_campania']],
                        on='producto', how='left')
sin_campania = sin_campania.rename(columns={'dias_fuera_campania': 'dias_campania'})
sin_campania['canal'] = 'Sin campaña'
sin_campania['indice_venta_dia'] = sin_campania['importe_total'] / sin_campania['dias_campania']

# === 9. Combinar resultados ===
comparativo = pd.concat([con_campania, sin_campania], ignore_index=True)

# === 10. Listado completo con días evaluados ===
print("\n=== LISTADO COMPLETO POR PRODUCTO, CANAL Y DÍAS EVALUADOS ===")
print(
    comparativo[['producto', 'canal', 'dias_campania', 'cantidad', 'importe_total', 'indice_venta_dia']]
    .sort_values(['producto', 'canal'])
    .round(2)
)
print(f"\nTotal de días analizados para todo el periodo: {dias_periodo_total}")



Periodo analizado: 2024-01-02 a 2024-12-30 (364 días)
Ventas con campaña: 768
Ventas sin campaña: 2230

=== LISTADO COMPLETO POR PRODUCTO, CANAL Y DÍAS EVALUADOS ===
            producto        canal  dias_campania  cantidad  importe_total  \
0    Adorno De Pared        Email             28     187.0       15037.90   
1    Adorno De Pared         Rrss             61     187.0       15037.90   
90   Adorno De Pared  Sin campaña            230     446.0       33055.59   
2    Adorno De Pared           Tv             45     187.0       15037.90   
3           Alfombra        Email             36     224.0       15137.77   
..               ...          ...            ...       ...            ...   
86            Tablet           Tv             58     135.0        8744.12   
87         Televisor        Email             61     148.0       10827.04   
88         Televisor         Rrss             43     148.0       10827.04   
119        Televisor  Sin campaña            202     484.0      

Tabla comparativa de canales

In [156]:

# === 10. Obtener canales dinámicamente ===
canales = sorted(mktg_etl['canal'].dropna().unique().tolist() + ['Sin campaña'])

# === 11. Crear tabla comparativa ===
pivot_indice = comparativo.pivot_table(
    index='producto',
    columns='canal',
    values='indice_venta_dia',
    aggfunc='sum'
).reset_index()

# Asegurar columnas para todos los canales existentes
for canal in canales:
    if canal not in pivot_indice.columns:
        pivot_indice[canal] = 0

pivot_indice = pivot_indice[['producto'] + canales]

print("\n=== INDICE DE VENTA DIARIA ($/día) POR PRODUCTO Y CANAL ===")
print(pivot_indice.round(2))

# === 12. Promedio general por canal ===
promedios = comparativo.groupby('canal')['indice_venta_dia'].mean().round(2)
print("\n=== PROMEDIO GENERAL DEL INDICE DIARIO ($/día) POR CANAL ===")
print(promedios)

# === 13. Conclusión básica ===
canal_top = promedios.idxmax()
valor_top = promedios.max()
print(f"\nEl canal con mayor índice promedio diario de ventas es '{canal_top}' con ${valor_top:.2f} por día.")



=== INDICE DE VENTA DIARIA ($/día) POR PRODUCTO Y CANAL ===
canal                producto   Email    Rrss  Sin campaña      Tv
0             Adorno De Pared  537.07  246.52       143.72  334.18
1                    Alfombra  420.49  352.04       119.98  398.36
2                  Aspiradora  256.90  401.40       140.53  755.58
3                 Auriculares  356.49  568.16       219.59  699.27
4                    Batidora  391.01  617.38       125.09  510.01
5                    Cafetera  400.68  472.23       190.88  220.38
6                  Candelabro  185.71  153.21        19.38  185.71
7      Consola De Videojuegos  299.44  389.27       128.82  389.27
8                    Cortinas  311.84  488.55       122.80  357.47
9           Cuadro Decorativo  665.30  494.71       141.72  393.75
10             Cámara Digital  371.16  247.44       141.92  296.93
11      Elementos De Cerámica  351.91  387.10       131.25  221.20
12          Espejo Decorativo  777.56  243.94       133.11  319.00
1

**ANALISIS** El análisis conjunto de los registros de ventas y las campañas de marketing del período enero a diciembre de 2024 (364 días) evidencia una correlación positiva entre la ejecución de campañas y el aumento de las ventas diarias promedio.

Las campañas distribuidas en los distintos canales (Email, RRSS y TV) lograron incrementar significativamente el rendimiento diario respecto de las ventas sin apoyo de marketing:

Se graban las tablas en la subcarpeta de reportes


In [157]:
# === GUARDADO DE RESULTADOS FINALES ===
import os

# Crear carpeta /reportes si no existe
ruta_reportes = os.path.join(ruta_base, "reportes")
os.makedirs(ruta_reportes, exist_ok=True)

# 1. Guardar comparativo base (detalle con días, cantidades, etc.)
comparativo.to_csv(
    f"{ruta_reportes}/comparativo_ventas_producto_canal.csv",
    index=False,
    encoding="utf-8"
)

# 2. Guardar tabla de índice diario por canal
pivot_indice.to_csv(
    f"{ruta_reportes}/ventas_indice_diario_por_canal.csv",
    index=False,
    encoding="utf-8"
)

# 3. Guardar promedio general por canal
promedios.to_csv(
    f"{ruta_reportes}/ventas_promedio_general_por_canal.csv",
    header=["indice_venta_dia_promedio"],
    encoding="utf-8"
)

print("\nArchivos guardados correctamente en:")
print(f"- {ruta_reportes}/comparativo_ventas_producto_canal.csv")
print(f"- {ruta_reportes}/ventas_indice_diario_por_canal.csv")
print(f"- {ruta_reportes}/ventas_promedio_general_por_canal.csv")



Archivos guardados correctamente en:
- /content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics/reportes/comparativo_ventas_producto_canal.csv
- /content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics/reportes/ventas_indice_diario_por_canal.csv
- /content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics/reportes/ventas_promedio_general_por_canal.csv



---


### Mostramos la ruta y los archivos finales

In [158]:
# Mostrar todas las rutas y subrutas dentro de la carpeta base
print(f"Contenido completo de la carpeta base:\n{ruta_base}\n")

for raiz, carpetas, archivos in os.walk(ruta_base):
    print(f"📂 Carpeta: {raiz}")
    for carpeta in carpetas:
        print(f"   ├── 📁 {carpeta}")
    for archivo in archivos:
        print(f"   ├── 📄 {archivo}")
    print()

Contenido completo de la carpeta base:
/content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics

📂 Carpeta: /content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics
   ├── 📁 originales
   ├── 📁 Datos Limpios
   ├── 📁 reportes
   ├── 📄 CRUZ pre_entrega.ipynb

📂 Carpeta: /content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics/originales
   ├── 📄 ventas.csv
   ├── 📄 marketing.csv
   ├── 📄 clientes.csv

📂 Carpeta: /content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics/Datos Limpios
   ├── 📄 ventas_etl.csv
   ├── 📄 clientes_etl.csv
   ├── 📄 mktg_etl.csv

📂 Carpeta: /content/drive/MyDrive/Adriana Cruz - Comision 25262 - TPI Data Analytics/reportes
   ├── 📄 ventas_totales_mensuales.csv
   ├── 📄 ventas_top80_completo.csv
   ├── 📄 ventas_por_categoria.csv
   ├── 📄 comparativo_ventas_producto_canal.csv
   ├── 📄 ventas_indice_diario_por_canal.csv
   ├── 📄 ventas_promedio_general_por_canal.csv

